feat(angular): deprecate IonicModule (#31270)
driftyco/ionic: packages/angular/README.md

This is where the lazy loaded component implementations live.

Ionic developers can access this by importing from `@ionic/angular/lazy`.

> [!CAUTION]

> > The lazy loaded build and its `IonicModule` are deprecated and will be removed in a future major version. New code should use the standalone components and `provideIonicAngular()` imported from `@ionic/angular`.

Reviewer: ```suggestion
> The lazy loaded build, including `IonicModule`, is deprecated and will be removed in a future major version. New code should use the standalone components and `provideIonicAngular()` imported from `@ionic/angular`.
```

Author: https://github.com/ionic-team/ionic-framework/pull/31270/changes/254ca2f45d5f31bbf905d8e68ca0974e1e456f47
